File: UtilitiesVB/To Depricate/FileIO2.vb
Imports System.IO
Imports System.Linq

Public Module FileIO2
    Public Sub DELETE_TextFile(filename As String, stagingPath As String)
        Dim filepath As String = Path.Combine(stagingPath, filename)

        If File.Exists(filepath) Then
            File.Delete(filepath)
        End If

    End Sub

    <Flags>
    Private Enum WriteOptions
        None = 0
        AppendNewLine = 1
        OpenAsAppend = 2
    End Enum

    Public Sub Write_TextFile(strFileName As String, strOutput() As String, strFileLocation As String)
        Dim filepath As String = Path.Combine(strFileLocation, strFileName)
        Dim listOutput As New List(Of String)(strOutput)
        For Each output In listOutput
            WriteUTF8(filepath, output, WriteOptions.AppendNewLine AndAlso WriteOptions.OpenAsAppend)
        Next

    End Sub

    'Public Sub Write_TextFile(strFileName As String,
    '                          enumerableOutput As IEnumerable,
    '                          strFileLocation As String,
    '                          encoding As Text.Encoding)
    '    Dim filepath As String = Path.Combine(strFileLocation, strFileName)
    '    Dim listOutput As New List(Of String)(strOutput)
    '    WriteUTF8(filepath, listOutput, WriteOptions.AppendNewLine AndAlso WriteOptions.OpenAsAppend)
    'End Sub


    Private Sub WriteUTF8(filepath As String,
                          textString As String,
                          options As WriteOptions)

        Dim asAppend As Boolean = options.HasFlag(WriteOptions.OpenAsAppend)

        Using sw As StreamWriter = New StreamWriter(
            filepath, asAppend, System.Text.Encoding.UTF8)
            If options.HasFlag(WriteOptions.AppendNewLine) Then
                sw.WriteLine(textString)
            Else
                sw.Write(textString)
            End If
            sw.Close()
        End Using

    End Sub



    Public Function CSV_ReadTxtF(filename As String, fileaddress As String, Optional SkipHeaders As Boolean = True) As String()

        Dim filepath As String = Path.Combine(fileaddress, filename)

        If File.Exists(filepath) Then
            If SkipHeaders Then
                Dim lines = File.ReadAllLines(filepath)
                Return lines.Skip(1).ToArray()
            Else
                Return File.ReadAllLines(filepath)
            End If

        Else
            Return Nothing
        End If

    End Function

    Public Function CSV_Read(filename As String, fileaddress As String, Optional SkipHeaders As Boolean = False) As String()

        Dim filepath As String = Path.Combine(fileaddress, filename)

        If File.Exists(filepath) Then
            If SkipHeaders Then
                Dim lines = File.ReadAllLines(filepath, System.Text.Encoding.UTF8)
                Return lines.Skip(1).ToArray()
            Else
                Return File.ReadAllLines(filepath)
            End If

        Else
            Return Nothing
        End If

    End Function

    Public Function CSV_SPLIT_TO_2D(str1D() As String, Optional Delimeter As String = ",", Optional zerobased As Boolean = False) As String(,)
        Dim i, j As Integer
        Dim Count As Integer
        Dim maxj As Integer

        Dim strD2_tmp(,) As String
        Dim strTmp() As String
        Dim strLine As String
        Dim intBase As Integer

        If zerobased Then
            intBase = 0
        Else
            intBase = 1
        End If

        For i = LBound(str1D) To UBound(str1D)
            strLine = str1D(i)
            Count = Len(strLine) - Len(Replace(strLine, Delimeter, ""))
            If Count > maxj Then maxj = Count
        Next i

        ReDim strD2_tmp(UBound(str1D) + intBase, maxj + intBase)

        For i = 0 To UBound(str1D)
            strTmp = Split(str1D(i), Delimeter)
            For j = 0 To UBound(strTmp)
                strD2_tmp(i + intBase, j + intBase) = strTmp(j)
            Next j
        Next i

        Return strD2_tmp

    End Function

End Module

File: UtilitiesVB/To Depricate/FileIO2.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace UtilitiesVB
{

    public static class FileIO2
    {
        public static void DELETE_TextFile(string filename, string stagingPath)
        {
            string filepath = Path.Combine(stagingPath, filename);

            if (File.Exists(filepath))
            {
                File.Delete(filepath);
            }

        }

        [Flags]
        private enum WriteOptions
        {
            None = 0,
            AppendNewLine = 1,
            OpenAsAppend = 2
        }

        public static void Write_TextFile(string strFileName, string[] strOutput, string strFileLocation)
        {
            string filepath = Path.Combine(strFileLocation, strFileName);
            var listOutput = new List<string>(strOutput);
            foreach (var output in listOutput)
                WriteUTF8(filepath, output, (WriteOptions.AppendNewLine | WriteOptions.OpenAsAppend));
            
            
        }

        private static void WriteUTF8(string filepath, string textString, WriteOptions options)
        {

            bool asAppend = options.HasFlag(WriteOptions.OpenAsAppend);

            using (var sw = new StreamWriter(filepath, asAppend, System.Text.Encoding.UTF8))
            {
                if (options.HasFlag(WriteOptions.AppendNewLine))
                {
                    sw.WriteLine(textString);
                }
                else
                {
                    sw.Write(textString);
                }
                sw.Close();
            }

        }

        public static string[] CSV_ReadTxtF(string filename, string fileaddress, bool SkipHeaders = true)
        {

            string filepath = Path.Combine(fileaddress, filename);

            if (File.Exists(filepath))
            {
                if (SkipHeaders)
                {
                    string[] lines = File.ReadAllLines(filepath);
                    return lines.Skip(1).ToArray();
                }
                else
                {
                    return File.ReadAllLines(filepath);
                }
            }

            else
            {
                return null;
            }

        }

        public static string[] CSV_Read(string filename, string fileaddress, bool SkipHeaders = false)
        {

            string filepath = Path.Combine(fileaddress, filename);

            if (File.Exists(filepath))
            {
                string[] lines = File.ReadAllLines(filepath, System.Text.Encoding.UTF8);
                if (SkipHeaders)
                {
                    return lines.Skip(1).ToArray();
                }
                else
                {
                    return lines;
                }
            }

            else
            {
                return null;
            }

        }

        public static string[,] CSV_SPLIT_TO_2D(string[] str1D, string Delimeter = ",", bool zerobased = false)
        {
            int i, j;
            int Count;
            var maxj = default(int);

            string[,] strD2_tmp;
            string[] strTmp;
            string strLine;
            int intBase;

            if (zerobased)
            {
                intBase = 0;
            }
            else
            {
                intBase = 1;
            }

            var loopTo = Information.UBound(str1D);
            for (i = Information.LBound(str1D); i <= loopTo; i++)
            {
                strLine = str1D[i];
                Count = Strings.Len(strLine) - Strings.Len(Strings.Replace(strLine, Delimeter, ""));
                if (Count > maxj)
                    maxj = Count;
            }

            strD2_tmp = new string[Information.UBound(str1D) + intBase + 1, maxj + intBase + 1];

            var loopTo1 = Information.UBound(str1D);
            for (i = 0; i <= loopTo1; i++)
            {
                strTmp = Strings.Split(str1D[i], Delimeter);
                var loopTo2 = Information.UBound(strTmp);
                for (j = 0; j <= loopTo2; j++)
                    strD2_tmp[i + intBase, j + intBase] = strTmp[j];
            }

            return strD2_tmp;

        }

    }
}